pixy/httpapi_test.go

```diff
@@ -0,0 +1,173 @@
+package pixy
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	. "github.com/mailgun/kafka-pixy/Godeps/_workspace/src/gopkg.in/check.v1"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+type HTTPAPISuite struct {
+	addr       string
+	httpClient *http.Client
+	as         *HTTPAPIServer
+	errorCh    <-chan error
+	produced   []*production
+}
+
+var _ = Suite(&HTTPAPISuite{})
+
+func (s *HTTPAPISuite) SetUpSuite(c *C) {
+	InitTestLog()
+}
+
+func (s *HTTPAPISuite) SetUpTest(c *C) {
+	s.addr = path.Join(os.TempDir(), "kafka-pixy.sock")
+	s.httpClient = NewUDSHTTPClient(s.addr)
+	var err error
+	s.as, err = SpawnHTTPAPIServer(NetworkUnix, s.addr, s)
+	if err != nil {
+		panic(err)
+	}
+	s.produced = nil
+}
+
+func (s *HTTPAPISuite) TearDownTest(c *C) {
+	if s.as != nil {
+		s.as.Stop()
+		<-s.as.ErrorCh()
+	}
+}
+
+func (s *HTTPAPISuite) TestInvalidServerAddress(c *C) {
+	// When
+	as, err := SpawnHTTPAPIServer(NetworkUnix, "/tmp", s)
```

Comment:
[FHS](http://www.pathname.com/fhs/pub/fhs-2.3.html#VARRUNRUNTIMEVARIABLEDATA) states that /var/run/ is the canonical place for the Unix socket and pid file.

I'm guessing that we'll put it in /var/mailgun for our internal usage.